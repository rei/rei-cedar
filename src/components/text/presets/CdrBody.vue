<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import { baseTextProps } from '../../../types/interfaces';

defineOptions({
  name: 'CdrBody',
});

interface bodyTextProps extends baseTextProps {
   /** 
   * Sets the type scale
   * @type scaleValue
   * @values 0, 1
   */
  scale?: '0'|'1',
  /** Toggles the strong variant */
  strong?: boolean,
}

const props = withDefaults(defineProps<bodyTextProps>(), {
  tag: 'p',
  scale: '0',
  strong: false,
});

const typeProperties = computed(() => {
  return {
    '--cdr-body-font-size': `var(--cdr-type-scale-${props.scale})`,
    '--cdr-body-line-height': 
      `calc(var(--cdr-type-scale-${props.scale}) * var(--cdr-body-line-height-ratio))`,
    '--cdr-body-font-weight': props.strong ? '600' : '400',
  }
});

const baseClass = 'cdr-body';
const style = useCssModule();
</script>

<template>
  <component
    :is="tag"
    :class="style[baseClass]"
    :style="typeProperties"
  >
    <slot />
  </component>
</template>

<style module lang="scss">
@import "./styles/CdrPresets.module.scss";

.cdr-body {
  @include cdr-text-base-mixin;
  @include cdr-text-body-400;
  font-size: var(--cdr-body-font-size);
  line-height: var(--cdr-body-line-height);
  font-weight: var(--cdr-body-font-weight);
}
</style>  