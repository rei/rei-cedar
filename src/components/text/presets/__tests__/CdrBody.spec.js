import { mount } from '../../../../../test/vue-jest-style-workaround.js';
import CdrBody from '../CdrBody.vue';


describe('CdrBody', () => {
  describe('default configuration', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrBody, {
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('passing a scale', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrBody, {
        props: { scale: '1' },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('setting to strong variant', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrBody, {
        props: { strong: true },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});
