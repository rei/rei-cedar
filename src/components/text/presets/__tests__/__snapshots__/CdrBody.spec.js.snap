// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrBody > default configuration > renders correctly 1`] = `
<p
  class="cdr-body"
  style="--cdr-body-font-size: var(--cdr-type-scale-0); --cdr-body-line-height: var(--cdr-line-height-ratio-body-0); --cdr-body-font-weight: 400;"
>
  
  paragraph
  
</p>
`;

exports[`CdrBody > passing a scale > renders correctly 1`] = `
<p
  class="cdr-body"
  style="--cdr-body-font-size: var(--cdr-type-scale-1); --cdr-body-line-height: var(--cdr-line-height-ratio-body-1); --cdr-body-font-weight: 400;"
>
  
  paragraph
  
</p>
`;

exports[`CdrBody > setting to strong variant > renders correctly 1`] = `
<p
  class="cdr-body"
  style="--cdr-body-font-size: var(--cdr-type-scale-0); --cdr-body-line-height: var(--cdr-line-height-ratio-body-0); --cdr-body-font-weight: 600;"
>
  
  paragraph
  
</p>
`;
